### ISSUE TITLE
[Performance Problems] Pehkui

### ISSUE THREAD

**Issue by Virtuoel** [Virtuoel/Pehkui#375]
Report performance issues you have with Pehkui here.

It is highly recommended to add a profiling report in your comment.
You can create one by using the Spark mod and its profiler commands.
For more info, check these links:
https://spark.lucko.me/docs/Command-Usage
https://modrinth.com/mod/spark
https://www.curseforge.com/minecraft/mc-mods/spark

If your performance issue is specifically when Pehkui is used with another mod, add a comment to the corresponding issue from this list. If the mod has no issue yet, create a new issue and I'll add it to the list:
- https://github.com/Virtuoel/Pehkui/issues/376
- No other issues yet

**Comment by embeddedt** [comment]
Here is a Spark profiler report (not by me) showing entity tick times being greatly exacerbated by Pehkui. As you can see from the bookmarks, the majority of the time appears to be spent by the mixin that modifies bounding boxes retrieving the scale using `ScaleUtils`.

https://spark.lucko.me/pLFykk8HZU?hl=483,332

**Comment by Virtuoel** [comment]
Looking through that profile, it shows 99% of the time coming from `ItemEntity` collision checking, which to me sounds like some contraption is overflowing way too many items onto the ground.

**Comment by embeddedt** [comment]
This is true, but in any case Pehkui is clearly adding enough overhead to that to be visible in the profiler.

**Comment by JosTheDude** [comment]
This is months past the original post of this issue, but Pekhui is the biggest performance hit on my server with almost 4% of the total 8% of the thread that is being taken up being done by grabbing scale data; couldn't this be done async?

![image](https://github.com/Virtuoel/Pehkui/assets/43651265/7e1cb619-397f-46e2-a9df-0d337c5181e6)

https://spark.lucko.me/9lwL919s2Z

**Comment by 1356RYAN** [comment]
Im also having similar things happening
![image](https://github.com/Virtuoel/Pehkui/assets/64508604/f30ebf00-2883-49f7-90da-7dc3c565acc5)
https://spark.lucko.me/wv23LsFXCR

**Comment by Freyr-VM** [comment]
I'm having hella lag, it seems to be pehkui but im particulary new to this so i could be wrong 
![image](https://github.com/Virtuoel/Pehkui/assets/163369571/ef6c30e9-d6fe-40b1-8d1c-97e3f95632b5)
https://spark.lucko.me/eC6NVNsq1h

**Comment by souldbminerr** [comment]
Same

**Comment by VerdantVerger** [comment]
Same, working with bewitchment and Improved mobs 1 20 1 fabric
![imaаge](https://github.com/Virtuoel/Pehkui/assets/168965417/1d478b1d-ec80-4e06-bdad-683809aa7c93)

**Comment by miyoshikai** [comment]
Only during the Rain and Thunder weather, Pehkui's server thread usage is very high and lags the server.
On any other weather, clear, rain, there is no lag at all.
https://spark.lucko.me/xRuK4qFLAn

Deceasedcraft modpack 

Here is a spark link for when it is not rain and thunder
https://spark.lucko.me/gWiGETQvUL

ISSUE SOLVED: Foxes were the issue smh
https://bugs.mojang.com/browse/MC-179916

**Comment by nebulazorua** [comment]
> I'm having hella lag, it seems to be pehkui but im particulary new to this so i could be wrong ![image](https://github.com/Virtuoel/Pehkui/assets/163369571/ef6c30e9-d6fe-40b1-8d1c-97e3f95632b5) https://spark.lucko.me/eC6NVNsq1h

Having the same results when running Spark. tanking TPS from 20 to 9
